src/main/java/uk/gov/justice/digital/delius/controller/secure/AppointmentBookingController.java

```diff
@@ -71,4 +73,25 @@
         AppointmentCreateResponse response = appointmentService.createAppointment(crn, sentenceId, contextName, contextlessAppointmentCreateRequest);
         return new ResponseEntity<>(response, HttpStatus.CREATED);
     }
+
+    @RequestMapping(value = "/offenders/crn/{crn}/appointments/{appointmentId}/context/{contextName}",
+        method = RequestMethod.PATCH,
+        consumes = "application/json")
+    @ApiResponses(
+        value = {
+            @ApiResponse(code = 200, message = "Updated", response = String.class),
+            @ApiResponse(code = 400, message = "Invalid request", response = ErrorResponse.class),
+            @ApiResponse(code = 403, message = "Requires role ROLE_COMMUNITY_INTERVENTIONS_UPDATE"),
+            @ApiResponse(code = 500, message = "Unrecoverable error whilst processing request.", response = ErrorResponse.class)
+        })
+
+    @ApiOperation(value = "Updates an Contact appointment")
+    public ResponseEntity<AppointmentUpdateResponse> patchAppointmentWithContext(final @PathVariable("crn") String crn,
+                                                                                 final @PathVariable("appointmentId") Long appointmentId,
+                                                                                 final @PathVariable("contextName") String context,
+                                                                                 final @RequestBody JsonPatch jsonPatch) {
+
+        AppointmentUpdateResponse response = appointmentService.patchAppointment(crn, appointmentId, context, jsonPatch);
+        return new ResponseEntity<>(response, HttpStatus.OK);
+    }
 }
```

Comment:
No sure the advantage to this compared to say a PUT with a DTO that has `notifyPPOfAttendanceBehaviour`, `notes`, `attended` or whatever. Feels this requires a greater cognitive load and harder to document.

But happy to see how this plays out

Reply:
Yes introducing PATCH methods is something we discussed a lot with the probation TAs and devs. We're using this pattern where the client doesn't have enough information to construct a full PUT. It's probably worth a proper discussion on the approach and reasoning if we can make some time.

Reply:
Now using POST operation with DTO